### PR TITLE
gdb: add 11.2 and compile patch for glibc 2.35

### PR DESCRIPTION
**Pull request by joequant**

### CODE REVIEW

File: var/spack/repos/builtin/packages/gdb/package.py [hunk 1]
```diff
@@ -62,6 +63,11 @@ class Gdb(AutotoolsPackage, GNUMirrorPackage):
     # https://sourceware.org/pipermail/gdb-patches/2021-February/176622.html
     patch('importlib.patch', when="@8.3.1:10.2 ^python@3.4:")
 
+    # patch to use decltype instead of typeof to fix compile on glibc 2.35
```
Comment (robertu94) [inline var/spack/repos/builtin/packages/gdb/package.py:66]: Thanks for your patch. Can you please provide some additional context on this. IE what system (OS) did you test on? Where does this patch come from? Is this a patch from upstream? If not Is it going to be merged into upstream GDB? Did you confirm that it still works if you build with older gcc with some test output? I appreciate your time and contributions

**Comment by robertu94** [comment]
@alalazo pinging you because I’m not sure who from the spack core team to ask. This PR is a change to GDB that actually depends on the glibc version. As far as I know this isn’t modeled in spack’s dependency model, is this something about to be fixed, or how should this be handled?

**Comment by alalazo** [comment]
> As far as I know this isn’t modeled in spack’s dependency model, is this something about to be fixed, or how should this be handled?

`glibc`, as you said, it's not modeled. We plan to extend the model so that we might be able to get that deep in the stack of software, but that's a long term goal. For the time being, if the patch is forward compatible (i.e. it doesn't hinder use with recent `glibc`) I would just patch `gdb` wherever needed.

**Comment by alalazo** [comment]
@joequant Can you please reply to https://github.com/spack/spack/pull/30126#discussion_r851802846 and other comments?

**Comment by alalazo** [comment]
Closing, since there was no reply and v11.2 has been added in the meanwhile.

**Pull request closed without merging**